fsspec/implementations/smb.py

```diff
@@ -0,0 +1,251 @@
+# -*- coding: utf-8 -*-
+"""
+This module contains SMBFileSystem class responsible for handling access to
+Windows Samba network shares by using package smbprotocol
+"""
+
+from stat import S_ISDIR, S_ISLNK
+import datetime
+import types
+import uuid
+
+import smbclient
+
+from .. import AbstractFileSystem
+from ..utils import infer_storage_options
+
+
+class SMBFileSystem(AbstractFileSystem):
+    """Allow reading and writing to Windows and Samba network shares.
+
+    When using `fsspec.open()` for getting a file-like object the URI
+    should be specified as this format:
+    `smb://workgroup;user:password@server:port/share/folder/file.csv`.
+
+    Example::
+        >>> import fsspec
+        >>> with fsspec.open('smb://myuser:mypassword@myserver.com/'
```

Comment:
but you *could* also do
```
with fsspec.open('/share/folder/file.csv',
        host="myserver.com",
        username="myuser",
        password="mypassword"):
```

Reply:
This is missing smb:// somewhere, right?

Reply:
Sorry, 
```
with fsspec.open('smb://share/folder/file.csv',
        host="myserver.com",
        username="myuser",
        password="mypassword"):
```
(although you can also give `protocol="smb"` instead)

Reply:
What exactly should change here? 
Should we keep the doc example? Change it? Extend it?